query-engine/connectors/sql-query-connector/src/query_builder/write.rs

```diff
@@ -161,17 +162,24 @@ pub fn update_many(
 
     let query = query.append_trace(&Span::current()).add_trace_id(trace_id);
     let columns: Vec<_> = ModelProjection::from(model.primary_identifier()).as_columns().collect();
-    let result: Vec<Query> = super::chunked_conditions(&columns, ids, |conditions| query.clone().so_that(conditions));
+    let result: Vec<Query> = super::chunked_conditions(&columns, ids, |conditions| {
+        query.clone().so_that(conditions.and(filter_condition.clone()))
+    });
 
     Ok(result)
 }
 
-pub fn delete_many(model: &ModelRef, ids: &[&SelectionResult], trace_id: Option<String>) -> Vec<Query<'static>> {
+pub fn delete_many(
+    model: &ModelRef,
+    ids: &[&SelectionResult],
+    filter_condition: ConditionTree<'static>,
+    trace_id: Option<String>,
+) -> Vec<Query<'static>> {
     let columns: Vec<_> = ModelProjection::from(model.primary_identifier()).as_columns().collect();
 
     super::chunked_conditions(&columns, ids, |conditions| {
```

Comment:
This chunking would not be needed anymore if we'd just use the given filter... Could we just skip the `IN` here and if it's too hard, not touch the first `SELECT` yet? This `IN` is problematic e.g. if you delete a large table.

Reply:
I would like to address that in later work.